### PR TITLE
Updates required for using Cubed-Sphere Observed Precip File

### PR DESCRIPTION
**Pull request by lltakacs**

### CODE REVIEW

**Comment by sdrabenh** [comment]
Requires companions PR [#169](https://github.com/GEOS-ESM/GEOSgcm_GridComp/pull/559)

File: GEOS_ChemEnvGridComp.F90 [hunk 1]
```diff
@@ -826,57 +821,23 @@ subroutine Run2 ( GC, IMPORT, EXPORT, CLOCK, RC )
        end do
     end if
 
-! Import precip from MOIST or read observed precip from a file. 
+! Import total precip from SURFACE (either model or observed precip)
 ! Export total, convective and non-convective precipitation.
 !--------------------------------------------------------------
-    call MAPL_GetPointer ( EXPORT, tprec,    'TPREC',     __RC__ )
-    call MAPL_GetPointer ( EXPORT, cn_prcp,  'CN_PRCP',   __RC__ )
+    call MAPL_GetPointer ( EXPORT,    tprec,    'TPREC',  __RC__ )
```
Comment (gmao-rreichle) [inline GEOS_ChemEnvGridComp.F90:827]: does this need to be 'PRECTOT' ??? (i.e., string used in export spec)
sorry if this is completely off base.

Comment (gmao-rreichle) [reply GEOS_ChemEnvGridComp.F90:827]: Never mind... I got confused with imports and exports, only just noticed line 832 below.

**Comment by lltakacs** [comment]
No, the export from ChemEnvGridComp is TPREC.  This is sent to the children of Chemistry.  It is loaded with the Total Precip as computed in Surface, i.e., either with the Model State or the Observed Precip state.

From: Rolf Reichle ***@***.***>
Sent: Friday, March 4, 2022 10:19 AM
To: GEOS-ESM/GEOSchem_GridComp ***@***.***>
Cc: Takacs, Lawrence L. (GSFC-610.1)[SCIENCE SYSTEMS AND APPLICATIONS INC] ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [GEOS-ESM/GEOSchem_GridComp] Updates required for using Cubed-Sphere Observed Predip File (PR #169)


@gmao-rreichle commented on this pull request.

________________________________

In GEOS_ChemEnvGridComp.F90<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSchem_GridComp%2Fpull%2F169%23discussion_r819660292&data=04%7C01%7Clawrence.l.takacs%40nasa.gov%7Cf9cc585c5f3949935c2908d9fdf2604c%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637820039703630086%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=PnxY%2F%2BG0lIi35CGYtrRn9GCP%2B0vabUYkU%2FFRsGce0rA%3D&reserved=0>:

>  ! Export total, convective and non-convective precipitation.

 !--------------------------------------------------------------

-    call MAPL_GetPointer ( EXPORT, tprec,    'TPREC',     __RC__ )

-    call MAPL_GetPointer ( EXPORT, cn_prcp,  'CN_PRCP',   __RC__ )

+    call MAPL_GetPointer ( EXPORT,    tprec,    'TPREC',  __RC__ )

does this need to be 'PRECTOT' ??? (i.e., string used in export spec)
sorry if this is completely off base.

-
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSchem_GridComp%2Fpull%2F169%23pullrequestreview-900356311&data=04%7C01%7Clawrence.l.takacs%40nasa.gov%7Cf9cc585c5f3949935c2908d9fdf2604c%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637820039703630086%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=tImIpSQhpimVXP7gMHMZ8qbhv7HHp2PMmFL9qE9dy7Q%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAMRE3K62IPOFYQKGSEHH243U6ISX3ANCNFSM5P5YEOBQ&data=04%7C01%7Clawrence.l.takacs%40nasa.gov%7Cf9cc585c5f3949935c2908d9fdf2604c%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637820039703630086%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=J7TxR7%2FUJ8N0C0ydOavlFqnQS5ia0v5DfHkNoBnDdvc%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Clawrence.l.takacs%40nasa.gov%7Cf9cc585c5f3949935c2908d9fdf2604c%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637820039703630086%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=uhsJBHHm516ByTuL1j4jCFpTFT85e7twk%2BTwp49L0NI%3D&reserved=0> or Android<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Clawrence.l.takacs%40nasa.gov%7Cf9cc585c5f3949935c2908d9fdf2604c%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637820039703630086%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=XMxwYPkngEmHzpFPo9wO6zek9xTgs3Mc29yHNPVBjXM%3D&reserved=0>.

**Review by adarmenov — APPROVED** [review]

**Review by mmanyin — APPROVED** [review]
This cleans up a file-read, improving the code

**Comment by mmanyin** [comment]
Note:  CTM will need to provide PRECTOT instead of TPREC

**Comment by lltakacs** [comment]
Mike, can you expand a bit on your comment about the concern with CTM?
Just to be clear, if no observational precip file is included in the AGCM.rc, then this is a zero-diff modification.

**Comment by amdasilva** [comment]
Ll,

  The CTM will need some overhaul to work with GOCART-2G. This needs to be coordinated with moving the scavenging code out of GF to its own separate component. As it stands, G2G does not perform any (internal) scavenging and as such cannot be run in CTM mode. (Yes, it could run, but without any scavenging.)

    Arlindo

From: Lawrence Takacs ***@***.***>
Reply-To: GEOS-ESM/GEOSchem_GridComp ***@***.***>
Date: Friday, March 18, 2022 at 10:28 AM
To: GEOS-ESM/GEOSchem_GridComp ***@***.***>
Cc: "DA SILVA, ARLINDO M (GSFC-6101)" ***@***.***>, Review requested ***@***.***>
Subject: [EXTERNAL] Re: [GEOS-ESM/GEOSchem_GridComp] Updates required for using Cubed-Sphere Observed Precip File (PR #169)


Mike, can you expand a bit on your comment about the concern with CTM?
Just to be clear, if no observational precip file is included in the AGCM.rc, then this is a zero-diff modification.

—
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSchem_GridComp%2Fpull%2F169%23issuecomment-1072464936&data=04%7C01%7Carlindo.m.dasilva%40nasa.gov%7C47f579f16e074e522a4708da08eba198%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637832105371337411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=rEkIFeeR3pkuEzXOVkyZIVoMZ7xum6zitqkGI2fB3Ks%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAH5LTY6SN2EANKLBRLHYTTDVASHKHANCNFSM5P5YEOBQ&data=04%7C01%7Carlindo.m.dasilva%40nasa.gov%7C47f579f16e074e522a4708da08eba198%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637832105371337411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=u1BA7n7K7vX9ZQttl%2F1LWiF3RmOr8uvnH6kVCP3bt%2BE%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Carlindo.m.dasilva%40nasa.gov%7C47f579f16e074e522a4708da08eba198%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637832105371337411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=%2FOtK4rWcWEM1Yn2YFEGl3OlsEal5PF0ineLbCJ2WJc4%3D&reserved=0> or Android<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Carlindo.m.dasilva%40nasa.gov%7C47f579f16e074e522a4708da08eba198%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637832105371337411%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=o8gPzRTTF%2BOkq%2FHgkRL5mqhvrVYZi4%2F7NmtKaNVJ0Gg%3D&reserved=0>.

**Comment by mmanyin** [comment]
Hi Larry,  the MAPL_AddImportSpec section used to import TPREC, and now it imports PRECTOT.  In order to satisfy this import change in CTM, I will just need to edit one line in the ExtData.rc file.  My comment was merely a reminder to myself.

**Comment by amdasilva** [comment]
there are flags in there that can tell it, even though running gocart, to run
it through a scavenging code. whether its the one we want or not (the one
from in gmi now) is another story....

andrea

-------------------------------
Andrea M. Molod, Ph.D.
Global Modeling and Assimilation Office
NASA Goddard Space Flight Center
Desk: 301-614-6845
-------------------------------


From: "DA SILVA, ARLINDO M (GSFC-6101)" ***@***.***>
Date: Friday, March 18, 2022 at 10:33 AM
To: GEOS-ESM/GEOSchem_GridComp ***@***.***>
Cc: "Molod, Andrea M. (GSFC-6101)" ***@***.***>
Subject: Re: [EXTERNAL] Re: [GEOS-ESM/GEOSchem_GridComp] Updates required for using Cubed-Sphere Observed Precip File (PR #169)

Ll,

  The CTM will need some overhaul to work with GOCART-2G. This needs to be coordinated with moving the scavenging code out of GF to its own separate component. As it stands, G2G does not perform any (internal) scavenging and as such cannot be run in CTM mode. (Yes, it could run, but without any scavenging.)

    Arlindo

From: Lawrence Takacs ***@***.***>
Reply-To: GEOS-ESM/GEOSchem_GridComp ***@***.***>
Date: Friday, March 18, 2022 at 10:28 AM
To: GEOS-ESM/GEOSchem_GridComp ***@***.***>
Cc: "DA SILVA, ARLINDO M (GSFC-6101)" ***@***.***>, Review requested ***@***.***>
Subject: [EXTERNAL] Re: [GEOS-ESM/GEOSchem_GridComp] Updates required for using Cubed-Sphere Observed Precip File (PR #169)


Mike, can you expand a bit on your comment about the concern with CTM?
Just to be clear, if no observational precip file is included in the AGCM.rc, then this is a zero-diff modification.

—
Reply to this email directly, view it on GitHub<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGEOS-ESM%2FGEOSchem_GridComp%2Fpull%2F169%23issuecomment-1072464936&data=04%7C01%7Candrea.m.molod%40nasa.gov%7Cd487d35348c54626bcc208da08ec36b6%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637832107864338775%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=IVI%2FWqOBwgZXJy1Rs6a6j7vHaPdkhyRgkus7aPHPpg4%3D&reserved=0>, or unsubscribe<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAH5LTY6SN2EANKLBRLHYTTDVASHKHANCNFSM5P5YEOBQ&data=04%7C01%7Candrea.m.molod%40nasa.gov%7Cd487d35348c54626bcc208da08ec36b6%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637832107864338775%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=Nr%2BDU5Hgfl253XXBjsZJT1bjBfLLvmHTvODqVIBJaEQ%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Candrea.m.molod%40nasa.gov%7Cd487d35348c54626bcc208da08ec36b6%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637832107864338775%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=q02W7PgpsZGUmQs3ii4f1K0qjo7QT2zv%2FjbiUncdLIY%3D&reserved=0> or Android<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Candrea.m.molod%40nasa.gov%7Cd487d35348c54626bcc208da08ec36b6%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C637832107864338775%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=UnWyd9Q0JhUsvhlfWeVW9B6hm%2FOFixeChmG8mGdUIOY%3D&reserved=0>.

**Pull request merged**